### PR TITLE
doc: release-notes-latest: Summarize networking changes from Zephyr

### PR DESCRIPTION
**Pull request by rlubos**
Update the changelog based on the Zephyr PR list from the recent
upmerge.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

### CODE REVIEW

**Comment by rlubos** [comment]
@carlescufi the `CI-doc-build` label apparently is not doing what it is expected to do - even though the label was when creating the PR, the `FULL_CI` plan stll tries to build the samples. Who can look into that?

**Comment by rlubos** [comment]
> > @carlescufi the `CI-doc-build` label apparently is not doing what it is expected to do - even though the label was added when creating the PR, the `FULL_CI` plan stll tries to build the samples. Who can look into that?
> 
> That label, `CI-doc-build`, should go away since the docs are being built by GitHub Actions, which are not affected by labels. So let's disregard this altogether.

I understand that docs are built by GH Actions now, but shouldn't the label be respected by the `FULL_CI` plan, i.e. skip all its tests if detected? Or do we just drop the functionality, and require to build samples for every single PR?

**Comment by carlescufi** [comment]
> > > @carlescufi the `CI-doc-build` label apparently is not doing what it is expected to do - even though the label was added when creating the PR, the `FULL_CI` plan stll tries to build the samples. Who can look into that?
> > 
> > 
> > That label, `CI-doc-build`, should go away since the docs are being built by GitHub Actions, which are not affected by labels. So let's disregard this altogether.
> 
> I understand that docs are built by GH Actions now, but shouldn't the label be respected by the `FULL_CI` plan, i.e. skip all its tests if detected? Or do we just drop the functionality, and require to build samples for every single PR?

I do not know the answer to that question. @shantha-14 or @thst-nordic should be able to answer that.

**Comment by thst-nordic** [comment]
I'll check it.

File: doc/nrf/releases/release-notes-latest.rst [hunk 1]
```diff
@@ -233,13 +247,20 @@ The following list summarizes the most important changes inherited from upstream
...
+    * Added a microsecond timer API implementation to the Zephyr port.
+
   * MQTT:
 
     * Fixed logging of UTF-8 strings.
 
   * Sockets:
 
     * Fixed TLS sockets access from user space.
+    * :option:`CONFIG_NET_SOCKETS_POSIX_NAMES` is now enabled by default.
```
Comment (ru-fu) [inline doc/nrf/releases/release-notes-latest.rst:260]:
```suggestion
    * Enabled :option:`CONFIG_NET_SOCKETS_POSIX_NAMES` by default.
```

File: doc/nrf/releases/release-notes-latest.rst [hunk 2]
```diff
@@ -233,13 +247,20 @@ The following list summarizes the most important changes inherited from upstream
...
+    * Added :c:macro:`TLS_DTLS_HANDSHAKE_TIMEOUT_MIN` and :c:macro:`TLS_DTLS_HANDSHAKE_TIMEOUT_MAX` TLS socket options.
+    * Fixed :c:func:`poll` behavior during handshake with DTLS socket.
```
Comment (ru-fu) [inline doc/nrf/releases/release-notes-latest.rst:262]:
```suggestion
    * Fixed :c:func:`zsock_poll` behavior during handshake with DTLS socket.
```
Correct?
If I understand it correctly, this function can also be exposed as poll(), but it's not possible to link to that name in the docs.

Comment (rlubos) [reply doc/nrf/releases/release-notes-latest.rst:262]: yes, `zsock_poll` is also fine if we can't use `poll`

**Review by ru-fu — APPROVED** [review]

**Comment by rlubos** [comment]
Outdated

**Pull request closed without merging**